src/ir/module.h

```diff
@@ -34,11 +34,13 @@ class Block;
 class Operation {
  public:
   Operation(const Block* parent, const Operator& op,
-            NamedAttributeMap attributes, NamedValueMap inputs)
+            NamedAttributeMap attributes, NamedValueMap inputs,
+            std::unique_ptr<Module> module = nullptr)
       : parent_(parent),
-        op_(&op),
+        op_(std::addressof(op)),
```

Comment:
Why `std::addressof` instead of just `&`? I've never had to use it before, it looks like it gets the actual address when `&` is overloaded. Is that case coming up here?

Reply:
It seemed more readable and cleaner. That is why I started shifting to this everywhere.